tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/PrimeRule.scala

```diff
@@ -21,8 +21,11 @@ class PrimeRule extends RewritingRule {
 
   override def apply(state: SymbState): SymbState = {
     state.ex match {
-      case OperEx(TlaActionOper.prime, NameEx(name)) =>
-        state.setRex(NameEx(name + "'"))
+      case OperEx(TlaActionOper.prime, nEx @ NameEx(name)) =>
+        nEx.typeTag match {
+          case Typed(tt1: TlaType1) => state.setRex(tla.name(name + "'", tt1))
+          case _                    => throw new InternalCheckerError("Internal expressions should be Typed", nEx)
+        }
```

Comment:
Do we actually need this extra check?

Feels like it only replicates the check in `TlaType1.fromTypeTag`, and if we wanted to be consistent, we'd need to replicate this kind of logic across all the rules?

Reply:
No, that's fair, we could just use that.